src/lowlevel/Types.jl

```diff
@@ -12,6 +12,11 @@ struct SoapySDRKwargs
     size::Csize_t
     keys::Ptr{Cstring}
     vals::Ptr{Cstring}
+    function SoapySDRKwargs(s,k,v)
+        this = new(s,k,v)
+        finalizer(SoapySDRKwargs_clear, this)
```

Comment:
this doesn't work. The struct here has no identity. The finalizer on the mutable struct was correct. Actually, I now realize that it was correct before, because the pointer to the julia object and the pointer to the C object are actually the same (because we inline the C struct by value). My apologies.

Reply:
Ace. I needed to do some more studying up on this anyway. Thanks for confirming what I was struggling to articulate :) I'll revert 5f8daa0 and this should be good to go?